data/scripts.js

```diff
@@ -1402,6 +1402,25 @@ exports.BattleScripts = {
 			moves[3] = toId(template.requiredMove);
 			hasMove[toId(template.requiredMove)] = true;
 		}
+		
+		// all viable aegislash sets should have king's shield
+		if (template.species === 'Aegislash' && !hasMove['kingsshield']) {
+			// replace a non-attacking move if the set has it
+			for (var i = 0; i < moves.length; ++i) {
+				if (moves[i].category === 'Status') {
```

Comment:
This doesn't work. You have to call `Tools#getMove` on the move in order to check its category. `moves[i]` is just a string.